roles/rsyslog/tasks/main.yml

```diff
@@ -66,46 +66,72 @@
         path: '{{ __rsyslog_work_dir }}'
         mode: '0700'
 
-    # Back up the pre-existing rsyslog config files in the backup dir,
-    # then removing the files/dirs in rsyslog.d.
+    # Back up the pre-existing rsyslog config files in the backup dir.
+    # If logging_purge_confs is set to true, remove files in rsyslog.d.
+    # Then restore the pre-existing config files.
+    # If logging_reset_confs is set to true, reinstall rsyslog package,
+    # and restore the pre-existing config files on top of it.
     - block:
         - name: Create a backup dir
           file:
             state: directory
             path: '{{ __rsyslog_backup_dir }}'
             mode: '0700'
 
-        - name: Get backup file stat
+        - name: Archive the additional contents in {{ __rsyslog_config_dir }}
+            to the backup dir, if any
+          shell: |
+            set -euo pipefail
+            rm -f "{{ __rsyslog_backup_dir }}/backup_additional.tgz"
+            for conf in $( ls "{{ __rsyslog_config_dir }}" ); do
+              echo $conf
+              rstr=$( rpm -qf "{{ __rsyslog_config_dir }}"/$conf 2>&1 || : )
+              if [[ $rstr != *"not owned by any package"* ]]; then
+                tar -cvzPf "{{ __rsyslog_backup_dir }}/backup_additional.tgz" \
+                  "{{ __rsyslog_config_dir }}"/$conf
+              fi
+            done
+
+        - name: Get stat of "{{ __rsyslog_backup_dir }}/backup_additional.tgz"
           stat:
-            path: "{{ __rsyslog_backup_dir }}/backup.tgz"
+            path: "{{ __rsyslog_backup_dir }}/backup_additional.tgz"
           register: __backup_stat
 
-        - name: Archive the original contents of {{ __rsyslog_config_dir }}
-            to the backup dir
-          command: >
-            tar -cvzPf "{{ __rsyslog_backup_dir }}/backup.tgz"
-              /etc/rsyslog.conf "{{ __rsyslog_config_dir }}"
-          when: not __backup_stat.stat.exists
+        - block:
+            - name: Purge original conf
+              file:
+                state: absent
+                path: "{{ __rsyslog_config_dir }}/*"
```

Comment:
Isn't this approach described as not working in #240?

Reply:
In #240, @richm says _"file module state=absent does not work with path=/some/dir/* (wildcard)"_. `And changes file: state: absent` to `shell: /bin/rm ...` in 
https://github.com/linux-system-roles/logging/pull/240/files#diff-a83292ff37790599fb1a73e2e3a0f69a0f699fdfa00d9af6f57a7b039ea1ddf1L91-L93
I wonder if `path: "{{ __rsyslog_config_dir }}/*"` works in your case.